specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+                description: "Array of values of the selected attribute, in the same corresponding order of the 'index' array."
+          examples:
+            application/json:
+              {
+                "entityId": "Kitchen1",
+                "attrName": "temperature",
+                "index": [1510826981388, 1510826990438, 1510826999725],
+                "values": [24.1, 25.3, 26.7]
+              }
+
+  # TODO: Could we remove all these /value suffixed endpoints if we provide the 'options' query arg? (keyvalues or values)?
+  /entities/{entityId}/attrs/{attrName}/value:
+    get:
+      operationId: reporter.reporter.query_1T1E1A_value
+      summary: "History of an attribute (values only) of a given entity instance."
+      description: "Similar to the previous, but focusing on the values regardles of the metadata."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+
+
+  /entities/{entityId}:
+    get:
+      operationId: reporter.reporter.query_1T1ENA
+      summary: "History of N attributes of a given entity instance."
+      description: "For example, query the average pressure, temperature and humidity of this month in the weather station WS1."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/attrs'
+        - $ref: '#/parameters/aggrMethod'
+        - $ref: '#/parameters/aggrPeriod'
+        - $ref: '#/parameters/options'
+        - $ref: '#/parameters/from'
+        - $ref: '#/parameters/to'
+        - $ref: '#/parameters/lastN'
+        - $ref: '#/parameters/limit'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+              attrName:
+                type: string
+              index:
+                type: array
+                items:
+                  type: number
+                description: "Array of the timestamps of the requested data. Values are UTC milliseconds since 1970-01-01T00:00:00Z."
+              values:
+                type: array
+                items:
+                  type: number
+                description: "Array of values of the selected attribute, in the same corresponding order of the 'index' array."
+          examples:
+            application/json:
+              {
+                "id": "Kitchen1",
+                "attrName": "temperature",
+                "index": [1510826981388, 1510826990438, 1510826999725],
+                "values": [24.1, 25.3, 26.7]
+              }
+
```

Comment:
what happens with two attributes or more attributes?

a good approach could be (and when for a time index there is not value for an attribute, we set it to null):

```
{
     "id": "Kitchen1",
     "index": [1510826981388, 1510826990438, 1510826999725],
     "attributes": [
          {
             "attrName": "temperature",
             "values": [24.1, 25.3, 26.7]
          },
          {
             "attrName": "pressure",
             "values": [1.01, 0.9, 1.02]
          },
     ]
}
```

Reply:
I was also thinking in this approach as the default, being a unique device, chances are that it pushes all data at once (so different measurements sharing the same timeset).

Reply:
agree, the problem comes with different entities, where the indexes may be totally different

Reply:
see comment on row 381